back/courses/scrapper.py

```diff
@@ -0,0 +1,105 @@
+
+    entry = soup.find_all("tr", attrs={"height": "30"})
+
+    today_range = [
+        timezone.make_aware(datetime.datetime(date.year, date.month, date.day, 0, 0)),
+        timezone.make_aware(datetime.datetime(date.year, date.month, date.day, 23, 59)),
+    ]
+
+    Timeslot.objects.filter(start__range=today_range).delete()
+
+    for course_entry in entry:
+        infos = course_entry.find_all("td")
+        start, end = map(
+            lambda h: parse_hours(date, h), infos[0].text.strip().split("-")
+        )
+        department = infos[1].text.strip()
+        emplacement = infos[2].text.split("-")[0].strip()
+        group = infos[3].text.strip()
+        group_number = int(group[3]) if group else None
+        if "-" not in infos[4].text:
+            acronym = ""
+            # name = infos[4].text.strip()
+        else:
+            acronym, _ = map(lambda s: s.strip(), infos[4].text.split("-", maxsplit=1))
+
+        if acronym and department != "DLC":
+            course_query = Course.objects.filter(acronym=acronym)
+            if course_query.exists():
+                course = course_query.first()
+
+                if (group_number is not None) or (
+                    group_number is None
+                    and not Timeslot.objects.filter(
+                        start__range=today_range, course_groups__course=course
+                    ).exists()
+                ):
+
+                    timeslot = Timeslot(
+                        start=start,
+                        end=end,
+                        place=emplacement,
+                    )
+                    timeslot.save()
+
+                    if group_number is not None:
+                        group_query = Group.objects.filter(
+                            course=course,
+                            number=group_number,
+                        )
+
+                        if group_query.exists():
+                            group = group_query.first()
+                        else:
+                            group = Group(
+                                course=course,
+                                teacher=course.teacher,
+                                number=group_number,
+                            )
+                            group.save()
+
+                        timeslot.course_groups.add(group)
+                    else:
+                        group_query = Group.objects.filter(
+                            course=course,
+                        )
+
+                        if group_query.exists():
+                            timeslot.course_groups.add(*group_query)
+                        else:
+                            group = Group(course=course, teacher=course.teacher)
+                            group.save()
+                            timeslot.course_groups.add(group)
+        else:
+            # Gérer les cours de DLC ici si on passe par emploidutemps
+            pass
```

Comment:
Why DLC courses need to be handled separately ?

Reply:
Because they have no acronym so it's hard to identify them